### PR TITLE
tlscommon: fix EC key decryption

### PR DESCRIPTION
**Pull request by AndersonQ**
## What does this PR do?

Fix how encrypted TLS certificate keys are handled. `x509.DecryptPEMBlock` returns a PEM block, and the code would assume it was not in PEM format and was re-encoding it into PEM, which would generate an invalid key.

## Why is it important?

Without the fix, it breaks when parsing a certificate key 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [ ]

## Related issues

- Relates https://github.com/elastic/elastic-agent/issues/2247
- Relates https://github.com/elastic/elastic-agent/issues/2248

### CODE REVIEW

**Comment by cmacknz** [comment]
Needs a test

File: transport/tlscommon/tls.go [hunk 1]
```diff
@@ -106,49 +105,49 @@ func ReadPEMFile(log *logp.Logger, s, passphrase string) ([]byte, error) {
...
-			blocks = append(blocks, &block)
+			// not in PEM format, encode it.
+			err = pem.Encode(buffer, &block)
+			if err != nil {
+				log.Errorf("Dropping encrypted pem block with private key, block type '%s' could not PEM encode it: %s", block.Type, err)
+				continue
+			}
 		default:
-			blocks = append(blocks, block)
+			// not encrypted and not in PEM format, encode it.
```
Comment (leehinman) [inline transport/tlscommon/tls.go:138]: Is this function only used for reading PEM encoded private keys?  The implementation suggests that, but the function name and signature don't.  If this were to read a PEM encoded certificate, you don't want to re-encode.

As it stands, I think the default case is trying to catch if they actually supplied a DER encoded private key.  If that is the case it is probably worth checking to see if `ParsePKIXPublicKey` works and to log at the warn level that DER format was provided but PEM format was expected.

**Review by leehinman — CHANGES_REQUESTED** [review]
needs tests

**Comment by AndersonQ** [comment]
`x509.DecryptPEMBlock` should receive a DER encoded key, not a PEM encoded. That way the key it returns will always be in DER format and we need to encode it as PEM. Thus the code is actually correct

**Pull request closed without merging**